modules/imgproc/src/median_blur.simd.hpp

```diff
@@ -489,7 +490,149 @@ medianBlur_8u_Om( const Mat& _src, Mat& _dst, int m )
+            v_uint16 v_ds = vx_load(hist + ofs);
+            for (int i = lanes; i <= step - lanes; i += lanes)
+                v_ds = v_add(v_ds, vx_load(hist + ofs + i));
+            ds = v_reduce_sum(v_ds);
+        }
+            // For smaller steps, use scalar accumulation
+        else
+#endif
+        {
+            ds = hist[ofs];
+            for (int i = 1; i < step; i++)
+                ds += hist[ofs + i];
+        }
+
+        // Determine if the cumulative sum has reached or surpassed half the total sum
+        m = (s + ds) <= halfsum;
+        ofs += m * step;
+        s += ds & -m;
+        step = step >> 1;
+    }
+}
+
+static void
+medianBlur_8u(const Mat &_src, Mat &_dst, int ksize) {
+    CV_INSTRUMENT_REGION();
+
+    const int channels = _src.channels();
+    int radius = ksize / 2;
+    CV_Assert(ksize % 2 == 1);
+
+    const int rows = _src.rows;
+    const int cols = _src.cols;
+    const int win_half_size = ksize * ksize / 2;
+
+    const uint8_t *src_data = _src.ptr<uint8_t>();
+    size_t src_step = _src.step;
+    uint8_t *dst_data = _dst.ptr<uint8_t>();
+    size_t dst_step = _dst.step;
+
+    double nstripes = (double) (rows * cols) / 1024.0;
+    parallel_for_(Range(0, rows), [&](const Range &r) {
+        int y_start = r.start;
+        int y_end = r.end;
+
+        std::vector<std::vector<uint16_t>> hist256(channels, std::vector<uint16_t>(256, 0));
+        std::vector<const uint8_t *> win_row_ptrs(ksize);
+
+        // initial hist for the first pixel
+        for (int dy = -radius; dy <= radius; dy++) {
+            int y = std::clamp(y_start + dy, 0, rows - 1);
+            const uint8_t *row_ptr = src_data + y * src_step;
+            win_row_ptrs[dy + radius] = row_ptr;
+
+            // always start from col 0, so need to add radius to first pixel
+            for (int c = 0; c < channels; ++c)
+                hist256[c][row_ptr[c]] += radius + 1;
+            // add the rest of the row
+            for (int dx = 1; dx <= radius; dx++) {
+                int x = std::min(dx, cols - 1);
+                const uint8_t *pixel_ptr = row_ptr + x * channels;
+                for (int c = 0; c < channels; ++c) {
+                    hist256[c][pixel_ptr[c]]++;
+                }
+            }
+        }
+
+        int x_start = 0, x_end = cols, x_step = 1;
+        // slide the window across the row
+        for (int y = y_start; y < y_end; y++) {
+            uint8_t *dst_pixel_ptr = dst_data + y * dst_step + x_start * channels;
+            for (int x = x_start; x != x_end; x += x_step, dst_pixel_ptr += x_step * channels) {
+                if (x != x_start) {
+                    // update hist by removing and adding pixel values as the window moves
+                    int x_remove = std::clamp(x - (radius + 1) * x_step, 0, cols - 1) * channels;
+                    int x_add = std::clamp(x + radius * x_step, 0, cols - 1) * channels;
+                    for (int dy = 0; dy < ksize; dy++) {
```

Comment:
please, create separate versions of histogram update loop for `channels == 1`, `channels == 3` and the generic case.